core/dtn7/src/routing/erouting/processing.rs

```diff
@@ -0,0 +1,255 @@
+    let broadcast_incoming = incoming.try_for_each(|msg| {
+        trace!(
+            "Received a external routing message: {}",
+            msg.to_text().unwrap().trim()
+        );
+
+        let packet: serde_json::Result<Packet> = serde_json::from_str(msg.to_text().unwrap());
+
+        match packet {
+            Ok(packet) => match packet {
+                // When a SenderForBundleResponse is received we check if a response channel for that
+                // bundle id exists and send the response on that channel.
+                Packet::SenderForBundleResponse(packet) => {
+                    trace!(
+                        "sender_for_bundle response: {}",
+                        msg.to_text().unwrap().trim()
+                    );
+
+                    if let Some(tx) = RESPONSES
+                        .lock()
+                        .unwrap()
+                        .remove(packet.bp.to_string().as_str())
+                    {
+                        if let Err(_) = tx.send(Packet::SenderForBundleResponse(packet)) {
+                            error!("sender_for_bundle response could not be passed to channel")
+                        }
+                    } else {
+                        info!("sender_for_bundle no response channel available")
+                    }
+                }
+                // Add a service on packet
+                Packet::ServiceAdd(packet) => {
+                    info!(
+                        "adding service via erouting {}:{}",
+                        packet.tag, packet.service
+                    );
+
+                    service_add(packet.tag, packet.service);
+                }
+                _ => {}
+            },
+            Err(err) => {
+                info!("err decoding external routing packet: {}", err);
+            }
+        }
+
+        future::ok(())
+    });
+
+    let receive_from_others = rx.map(Ok).forward(outgoing);
+
+    pin_mut!(broadcast_incoming, receive_from_others);
+    future::select(broadcast_incoming, receive_from_others).await;
+
+    info!("External routing disconnected");
+    disconnect();
+}
+
+fn disconnect() {
+    if CONNECTION.lock().unwrap().is_some() {
+        (*CONNECTION.lock().unwrap()) = None;
+    }
+}
+
+/// Sends a JSON encoded packet to the connected router.
+fn send_packet(p: &Packet) {
+    if let Ok(data) = serde_json::to_string(p) {
+        if let Some(con) = CONNECTION.lock().unwrap().as_ref() {
+            con.tx
+                .unbounded_send(Message::Text(data))
+                .expect("error while sending to tx");
+        }
+    }
+}
+
+/// Takes the RoutingNotification's, encodes them to serializable structs and then sends them
+/// to the external router if one is connected.
+pub fn notify(notification: RoutingNotifcation) {
+    let packet: Packet = match notification {
+        RoutingNotifcation::SendingFailed(bid, cla_sender) => {
+            Packet::SendingFailed(SendingFailed { bid, cla_sender })
```

Comment:
why not implement From trait for RoutingNotification in Packet?  would move the conversion to Packet and make the complexity of notify() smaller as its just the into() fall and then send_packet

Reply:
Done. Trait is implemented in Packet and erouting dev documentation was changed.